src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,158 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Storage class in an abstraction of a file loading and a saving system.
+ */
+public class Storage {
+    public static final String DEFAULT_STORAGE_FILE = "data/duke.txt";
+    public String filePath;
+    public File f;
+
+    public Storage(String filePath) throws DukeException {
+        if (!isValidFileExtension(filePath)) {
+            throw new DukeException("\t\tFile should end with .txt extension");
+        }
+        this.filePath = filePath;
+        this.f = new File(filePath);
+
+    }
+
+    /**
+     * Constucts an instance of the Storage class.
+     */
+    public Storage() {
+        this.filePath = "data";
+        this.f = new File(filePath, "duke.txt");
+        this.f.getParentFile().mkdirs();
+    }
+
+    /**
+     * Checks the validity of a filepath
+     * @param filePath the path of the file that is to be checked for validity
+     * @return boolean value stating is the file is a valid one
+     */
+    public boolean isValidFileExtension(String filePath) {
+        return filePath.endsWith(".txt");
+    }
+
+    /**
+     * Loads duke.txt file from the local disk.
+     * @return a list of tasks that is written in the duke.txt file
+     * @throws DukeException an exception when loading a file from the local disk
+     */
+    public ArrayList<Task> loadFile() throws DukeException {
+
+        ArrayList<Task> taskArr = new ArrayList<>();
+
+        try {
+            Scanner sc = new Scanner(this.f);
+            while (sc.hasNext()) {
+                taskArr.add(stringToTask(sc.nextLine()));
+            }
+        }
+        catch (FileNotFoundException e) {
+            throw new DukeException("File not found!");
+        }
+        return taskArr;
+
+    }
+
+    /**
+     * Saves a file to the disk with task details.
+     * @param TaskArr a list of tasks that needs to be saved to the disk
+     * @throws DukeException an exception when saving the file to the local disk
+     */
+    public void saveFile(ArrayList<Task> TaskArr) throws DukeException {
+        try {
+            FileWriter fw = new FileWriter(DEFAULT_STORAGE_FILE);
+            for (Task t : TaskArr) {
+                fw.write(taskToString(t) + "\n");
+            }
+            fw.close();
+        }
+        catch (IOException e) {
+            throw new DukeException("Error saving the Tasks into a file");
+        }
+    }
+
+    /**
+     * Converts strings to Task instances
+     * @param task the task string that need to be converted to a Task Object
```

Comment:
U should add a blank line after describing ur method and before the start of describing method headers.